pkg/status/combinedstatus-resolution.go

```diff
@@ -586,22 +617,22 @@ func handleCombinedFields(scName string, scData *statusCollectorData) *v1alpha1.
 	if len(scData.GroupBy) > 0 {
 		for _, groupByNamedExp := range scData.GroupBy {
 			groupByNameToValueToWorkStatus[groupByNamedExp.Name] = make(map[any][]*workStatusData)
+
+			for _, wsData := range scData.wecToData {
```

Comment:
A `workStatusData` is specific to a Binding, it does not apply across all Bindings that reference the same StatusCollector, right? If so then it would help if the comment on that type said so.